### PR TITLE
Rename the prefix of environment variables to NVFUSER.

### PR DESCRIPTION
**Pull request by naoyam**
Renamed `PYTORCH_NVFUSER_` to just `NVFUSER_`, e.g., `PYTORCH_NVFUSER_DUMP` is now `NVFUSER_DUMP`.

PYTORCH_NVFUSER is still available but a warning is issued if used.

### CODE REVIEW

**Comment by naoyam** [comment]
!build

File: test/test_dynamic_type.cpp [hunk 1]
```diff
@@ -291,7 +291,8 @@ TEST_F(DynamicTypeTest, Typing) {
   static_assert(!can_static_cast<IntSomeType, CustomType>);
   static_assert((int64_t)IntSomeType(1) == 1);
   EXPECT_THAT(
-      []() { (SomeType) IntSomeType(1); },
+      // suppress unused value warning
+      []() { (void)(SomeType)IntSomeType(1); },
```
Comment (naoyam) [inline test/test_dynamic_type.cpp:295]: clang issues an unused warning

**Comment by naoyam** [comment]
!build

**Review by zasdfgbnm — APPROVED** [review]
Thanks for the cleanup!

**Pull request merged**